sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -492,8 +494,9 @@ joint_matrix_store(Group sg,
       PI_ERROR_INVALID_DEVICE);
 #else
   // intel's impl
-  T *Ptr = dst.get();
-  __spirv_JointMatrixStoreINTEL<T, Tp, NumRows, NumCols,
+  using DecorT = typename sycl::detail::DecoratedType<T, Space>::type;
+  DecorT *Ptr = dst.get();
```

Comment:
This will sadly only work when `IsDecorated` is either `decorated::yes` or `decorated::legacy`. Instead you could do something like:
```suggestion
  DecorT *Ptr = [&]() {
     if constexpr (IsDecorated == access::decorated::legacy)
        return dst.get();
      else
        return dst.get_decorated();
    }();
```
or split it into a separate helper function (which I would probably prefer.) When we then one day get rid of the `legacy` variant we can just replace it with `dst.get_decorated()`.

Reply:
I split it into separate function